doc/flux_to_redux.md

```diff
@@ -0,0 +1,579 @@
+1. __Refactor the actions.__ Remove the action container class, and replace action instances with classes.
+    ```dart
+    // Before
+    class ExampleActions {
+      final Action<void> randomizeBackgroundColor = Action();
+      final Action<String> setBackgroundColor = Action();
+    }
+     
+    // After
+    class RandomizeBackgroundColorAction {}
+    class SetBackgroundColorAction {
+      final String backgroundColor;
+      SetBackgroundColorAction(this.backgroundColor);
+    }
+    ```
+
+    If your `Action` receives a custom class as typing for the action payload, that class could be a great starting point for the creation of your new Redux action.
+
+    ```dart
+    // Before
+    class ExampleActions {
+        final Action<SetBackgroundColorPayload> randomizeBackgroundColor = Action();
+    }
+
+    class SetBackgroundColorPayload {
+        String backgroundColor;
+        SetBackgroundColorAction(this.backgroundColor);
+    }
+    
+    // After
+    class SetBackgroundColorAction {
+        final String backgroundColor;
+        SetBackgroundColorAction(this.backgroundColor);
+    }
+    ```
+
+1. __Pull state mutation logic out of the store and into a reducer.__ Typically within a Flux store you have a `triggerOnActionV2` call that identifies an action and a function used to respond to that action. In many cases, that same logic should be perfect for a reducer.
+    ```dart
+    // Before transition
+    class ExampleFluxStore extends flux.Store {
+        FluxActions _actions;
+
+        String _backgroundColor = 'Gray';
+        String get backgroundColor = _backgroundColor;
+
+        // Constructor
+        ExampleFluxStore(this._actions) {
+            triggerOnActionV2(_actions.changeBackgroundColor, _changeBackgroundColor);
+        }
+
+        // Function to update the _backgroundColor state field
+        _changeBackgroundColor(_) {
+            _backgroundColor = '#' + (Random().nextDouble() * 16777215).floor().toRadixString(16);
+        }
+    }
+
+    // After being transitioned to a reducer...
+    ExampleState reducer(ExampleState oldState, dynamic /*or an action type*/ action) {
+        // Assumes an action called ChangeBackgroundColor was created in step 1
+        if (action is ChangeBackgroundColor) {
+            final color = '#' + (Random().nextDouble() * 16777215).floor().toRadixString(16);
+            return ExampleState.update(color: color);
+        }
+
+        return oldState;
+    }
+    ```
+1. __Create the state model.__ After pulling out state mutation logic, this should be as simple as renaming the class, not inheriting from the Flux store class, and cleaning up anything unrelated to the actual state fields. 
+1. __Instantiate the store class, using the state model and reducer.__
+1. __Add a `ReduxProvider` around the component tree.__ This will take in the store instantiated in step 4 as the component's `store` prop.
```

Comment:
Out of curiosity - why do none of these examples show any example code for the UI bits? (e.g. showing the actual wrapping of the component)

Reply:
All of the questions / points on adding UI examples are really good ones - my main reason was just that UI snippets were bulky and all of the code could be seen in the example. 

Adding the `ReduxProvider` wouldn't really be bulky though, so I added those examples!

For general "Refactoring UI Components", I just referred them to the examples in the web directory. I also added more comments in the Influx and Redux components to show what the refactor included. @aaronlademann-wf does that cover everything or do you think it'd be better in a markdown doc with the before and after examples together?